core/snippets/snippets_insert_raw_text.py

```diff
@@ -21,7 +21,7 @@
     desc="""If true, inserting snippets as raw text will always be done through pasting""",
 )
 
-RE_STOP = re.compile(r"\$(\d+|\w+)|\$\{(\d+|\w+)\}|\$\{(\d+|\w+):(.+)\}")
+RE_STOP = re.compile(r"\\?(?:\$(\d+|\w+)|\$\{(\d+|\w+)\}|\$\{(\d+|\w+):(.+)\})")
```

Comment:
This is now complicated enough that it probably warrants a comment breaking down the different parts.

Reply:
Adding comments also made me realize I messed this up. Not sure how it worked in my testing! Updated version incoming...

Reply:
Agreed! Commenting the regex made me realize I was doing unnecessary work when matching a `\$`.